core/src/main/java/org/keycloak/sdjwt/IssuerSignedJWT.java

```diff
@@ -218,21 +413,107 @@ public Builder withNestedDisclosures(boolean nestedDisclosures) {
+            this.jwsHeader.setType(jwsType);
+            return this;
+        }
+
+        public Builder withJwsHeader(JWSHeader jwsHeader) {
+            this.jwsHeader = jwsHeader;
+            return this;
+        }
+
+        public Builder withKid(String kid) {
+            getJwsHeader().setKeyId(kid);
+            return this;
+        }
+
+        public Builder withX5c(List<String> x5c) {
+            getJwsHeader().setX5c(x5c);
+            return this;
+        }
+
+        public Builder withX5c(String x5c) {
+            getJwsHeader().addX5c(x5c);
+            return this;
+        }
+
+        public Builder withX5c(Certificate x5c) {
+            getJwsHeader().addX5c(x5c);
+            return this;
+        }
+
+        public Builder withIat(long iat) {
+            getClaims().add(new VisibleSdJwtClaim(SdJwtClaimName.of(ClaimVerifier.CLAIM_NAME_IAT), new LongNode(iat)));
+            return this;
+        }
+
+        public Builder withNbf(long nbf) {
+            getClaims().add(new VisibleSdJwtClaim(SdJwtClaimName.of(ClaimVerifier.CLAIM_NAME_NBF), new LongNode(nbf)));
+            return this;
+        }
+
+        public Builder withExp(long exp) {
+            getClaims().add(new VisibleSdJwtClaim(SdJwtClaimName.of(ClaimVerifier.CLAIM_NAME_EXP), new LongNode(exp)));
+            return this;
+        }
+
+        /**
+         * this method requires the public key to be present in the keybindingJwts header as "jwk" claim
+         */
+        public Builder withKeyBinding(KeyBindingJWT keyBinding) {
+            ObjectNode cnf = JsonNodeFactory.instance.objectNode();
+            Optional.ofNullable(keyBinding.getJwsHeader().getOtherClaims().get("jwk"))
+                    .map(map -> JsonSerialization.mapper.convertValue(map, ObjectNode.class))
+                    .ifPresent(jwkNode -> cnf.set("jwk", jwkNode));
+            if (!cnf.isEmpty()) {
+                getClaims().add(new VisibleSdJwtClaim(SdJwtClaimName.of("cnf"), cnf));
+            }
+            return this;
+        }
+
+        public Builder withKeyBinding(JWK keyBinding) {
+            return withKeyBinding(JsonSerialization.mapper.convertValue(keyBinding, ObjectNode.class));
+        }
+
+        public Builder withKeyBinding(ObjectNode keyBinding) {
+            ObjectNode cnf = JsonNodeFactory.instance.objectNode();
+            cnf.set("jwk", keyBinding);
+            getClaims().add(new VisibleSdJwtClaim(SdJwtClaimName.of("cnf"), cnf));
+            return this;
+        }
+
+        public Builder withClaim(SdJwtClaim sdJwtClaim) {
+            getClaims().add(sdJwtClaim);
             return this;
         }
 
         public IssuerSignedJWT build() {
             // Preinitialize hashAlg to sha-256 if not provided
             hashAlg = hashAlg == null ? "sha-256" : hashAlg;
-            jwsType = jwsType == null ? "dc+sd-jwt" : jwsType;
+            jwsHeader.setType(jwsHeader.getType() == null ? "dc+sd-jwt" : jwsHeader.getType());
```

Comment:
This may also need some update once you rebase to avoid hardcoded constants from here.